### PR TITLE
Auto-detection of the project ID for gcloud.

### PR DESCRIPTION
**Pull request by james-lawrence**
can be autodetected from the environment on glcoud using the metadata
service.

### CODE REVIEW

File: providers/dns/gcloud/googlecloud_test.go [hunk 1]
```diff
@@ -48,6 +49,8 @@ func TestNewDNSProvider(t *testing.T) {
 			envVars: map[string]string{
 				"GCE_PROJECT":              "",
 				"GCE_SERVICE_ACCOUNT_FILE": "",
+				// as Travis run on GCE, we have to alter env
+				"GCE_METADATA_HOST": "http://lego.wtf",
```
Comment (james-lawrence) [inline providers/dns/gcloud/googlecloud_test.go]: hoboy.... that implies a security risk for travis....

Comment (ldez) [reply providers/dns/gcloud/googlecloud_test.go]: not really because the vars env are saved and restore between each tests.

Comment (james-lawrence) [reply providers/dns/gcloud/googlecloud_test.go]: i'd have to go play with it but if the GCE metadata is accessible in travis you can potentially pull ssh keys, instance metadata, etc. all of which can potentially be used to access travis' network.

**Comment by james-lawrence** [comment]
np, ty for the library. =)

**Pull request merged**